File: frontend/src/types/publication.ts
export type PublicationType = "journal_article" | "conference_paper" | "book_chapter" | "thesis" | "preprint" | "report" | "dataset";
export type PublicationStatus = "draft" | "under_review" | "revision_requested" | "accepted" | "published" | "rejected" | "withdrawn";
export type SubmissionStatus = "submitted" | "under_review" | "revision_required" | "revision_submitted" | "accepted" | "rejected" | "withdrawn";

export interface Publication {
  id: string;
  title: string;
  abstract: string | null;
  authors: string[];
  keywords: string[];
  pub_type: PublicationType;
  status: PublicationStatus;
  doi: string | null;
  version: number;
  project_id: string | null;
  submitter_id: string;
  created_at: string;
  updated_at: string;
}

export type PublicationCreate = Omit<Publication, "id" | "submitter_id" | "created_at" | "updated_at" | "version">;

export interface Submission {
  id: string;
  publication_id: string;
  venue_name: string;
  venue_type: string;
  status: SubmissionStatus;
  submitted_at: string;
  decision_at: string | null;
  decision_notes: string | null;
  submission_url: string | null;
  created_at: string;
}

export type SubmissionCreate = Omit<Submission, "id" | "publication_id" | "status" | "decision_at" | "decision_notes" | "created_at">;

File: backend/src/schemas/publication.py
from pydantic import BaseModel
from datetime import datetime
import uuid
from src.models.publication import PublicationType, PublicationStatus
from src.models.submission import SubmissionStatus


class PublicationCreate(BaseModel):
    title: str
    abstract: str | None = None
    authors: list[str] = []
    keywords: list[str] = []
    pub_type: PublicationType
    project_id: uuid.UUID | None = None


class PublicationUpdate(BaseModel):
    title: str | None = None
    abstract: str | None = None
    authors: list[str] | None = None
    keywords: list[str] | None = None
    status: PublicationStatus | None = None
    doi: str | None = None


class PublicationResponse(BaseModel):
    id: uuid.UUID
    title: str
    abstract: str | None
    authors: list[str]
    keywords: list[str]
    pub_type: PublicationType
    status: PublicationStatus
    doi: str | None
    version: int
    project_id: uuid.UUID | None
    submitter_id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SubmissionCreate(BaseModel):
    venue_name: str
    venue_type: str
    submitted_at: datetime
    submission_url: str | None = None


class SubmissionStatusUpdate(BaseModel):
    status: SubmissionStatus
    decision_notes: str | None = None
    decision_at: datetime | None = None


class SubmissionResponse(BaseModel):
    id: uuid.UUID
    publication_id: uuid.UUID
    venue_name: str
    venue_type: str
    status: SubmissionStatus
    submitted_at: datetime
    decision_at: datetime | None
    decision_notes: str | None
    submission_url: str | None
    created_at: datetime

    model_config = {"from_attributes": True}
